actor/src/main/scala-2.12/org/apache/pekko/util/ByteString.scala

```diff
@@ -790,6 +843,30 @@ sealed abstract class ByteString extends IndexedSeq[Byte] with IndexedSeqOptimiz
   // optimized in subclasses
   override def indexOf[B >: Byte](elem: B): Int = indexOf(elem, 0)
 
+  // optimized version of indexOf for bytes, optimized in subclasses
+  /**
+   * Finds index of first occurrence of some byte in this ByteString after or at some start index.
+   *
+   * Similar to indexOf, but it avoids boxing if the value is already a byte.
+   *
+   *  @param   elem   the element value to search for.
+   *  @param   from   the start index
+   *  @return  the index `>= from` of the first element of this ByteString that is equal (as determined by `==`)
+   *           to `elem`, or `-1`, if none exists.
+   */
+  def indexOf(elem: Byte, from: Int): Int = indexOf(elem, from)
```

Comment:
add `@since 1.1.0`

Reply:
How about add another `indexOf(elem:Byte,fromIndex:Int, toIndex:Int`?
or `def indexOf(fromIndex:Int, toIndex:Int, elem:Byte)`

Reply:
This mehtod is actually firstIndexOf or byteBefore

Reply:
Added the `@since` annotation in 44cf4bccf71cfaed828d81625c603ed1e66e3fab

Regarding adding more methods, I think that is a bit out of scope for this PR.

I was thinking a bit about that, and I think it would be nice if, when introducing the new methods, they were in sync with the stdlib.
For example, it would be pretty bad to introduce a `indexOf(elem:Byte,fromIndex:Int, toIndex:Int)` and then the stdlib introduce a `indexOf(elem:Byte, fromIndex:Int, lenght:Int)`.

So, I think one possible way to avoid that would be:
1. Port this PR to https://github.com/mdedetrich/bytestring
2. Try to move the `ByteString` to the stdlib - I think that was one of @mdedetrich main reasons for extracting the code (as discussed in https://contributors.scala-lang.org/t/adding-akkas-bytestring-as-a-scala-module-and-or-stdlib/5967), and I think this should now be possible under [SIP-51](https://github.com/scala/improvement-proposals/pull/54).
3. Propose the new methods to the stdlib, implementing the specialized version where it makes sense.

WDYT?

Reply:
Don't worry about the @mdedetrich project? This repo is the source of truth for ByteString. He will copy any changes from this repo if he thinks that they are useful.

The Pekko ByteString is not going away soon. The idea that it will end up in the Scala lib is not gaining traction and even it does it may not end up in every Scala version that Pekko needs it in and it may end up working differently to the way the Pekko one does.

Reply:
> The Pekko ByteString is not going away soon. The idea that it will end up in the Scala lib is not gaining traction and even it does it may not end up in every Scala version that Pekko needs it in and it may end up working differently to the way the Pekko one does.

It dis a really gain traction, it's just like qith most things you need to drive it yourself (others won't do it for you) and I didn't have the capacity with all of the other Pekko stuff going on

There is still a lot of merit in Pekko using an eventually stdlib `ByteString` but this will happen in a major breaking change in Pekko and it shouldn't dictate the development of Pekko.